h2o-extensions/xgboost/src/test/java/hex/tree/xgboost/XGBoostTest.java

```diff
@@ -2612,4 +2612,181 @@ public void testScalePosWeight() {
+      train.replace(train.find("Angaus"), train.vecs()[train.find("Angaus")].toCategoricalVec());
+      Scope.track(train);
+      DKV.put(train); // Update frame after hacking it
+
+      // Fix calibration set (the same way as training)
+      Scope.track(calib.vec("Angaus"));
+      calib.replace(calib.find("Angaus"), calib.vecs()[calib.find("Angaus")].toCategoricalVec());
+      Scope.track(calib);
+      DKV.put(calib); // Update frame after hacking it
+
+      XGBoostModel.XGBoostParameters parms = new XGBoostModel.XGBoostParameters();
+      parms._train = train._key;
+      parms._valid = train._key;
+      parms._response_column = "Angaus"; // Train on the outcome
+      parms._distribution = multinomial;
+      parms._ntrees = 5;
+      parms._tree_method = XGBoostModel.XGBoostParameters.TreeMethod.hist;
+      parms._seed = 42;
+      parms._col_sample_rate = 0.9;
+      model1 = new hex.tree.xgboost.XGBoost(parms).trainModel().get();
+      
+      parms._colsample_bylevel = 0.9;
+      model2 = new hex.tree.xgboost.XGBoost(parms).trainModel().get();
+      assert model1._output._training_metrics.rmse() == model2._output._training_metrics.rmse();
+    } finally {
+      Scope.exit();
+      if (model1 != null) model1.delete();
+      if (model2 != null) model2.delete();
+    }
+  }
+
+  @Test
+  public void testColSampleRateAndAlias() {
+    XGBoostModel model1 = null, model2 = null;
+    Scope.enter();
+    try {
+      Frame train = parseTestFile("smalldata/gbm_test/ecology_model.csv");
+      Frame calib = parseTestFile("smalldata/gbm_test/ecology_eval.csv");
+
+      // Fix training set
+      train.remove("Site").remove();     // Remove unique ID
+      train.remove("Method").remove();     // Remove unique ID
+      Scope.track(train.vec("Angaus"));
+      train.replace(train.find("Angaus"), train.vecs()[train.find("Angaus")].toCategoricalVec());
+      Scope.track(train);
+      DKV.put(train); // Update frame after hacking it
+
+      // Fix calibration set (the same way as training)
+      Scope.track(calib.vec("Angaus"));
+      calib.replace(calib.find("Angaus"), calib.vecs()[calib.find("Angaus")].toCategoricalVec());
+      Scope.track(calib);
+      DKV.put(calib); // Update frame after hacking it
+
+      XGBoostModel.XGBoostParameters parms = new XGBoostModel.XGBoostParameters();
+      parms._train = train._key;
+      parms._valid = train._key;
+      parms._response_column = "Angaus"; // Train on the outcome
+      parms._distribution = multinomial;
+      parms._ntrees = 5;
+      parms._tree_method = XGBoostModel.XGBoostParameters.TreeMethod.hist;
+      parms._seed = 42;
+      parms._col_sample_rate = 0.9;
+      parms._colsample_bylevel = 0.3;
+      model1 = new hex.tree.xgboost.XGBoost(parms).trainModel().get();
+      
+    } catch(H2OModelBuilderIllegalArgumentException ex){
+      assert ex.getMessage().contains("col_sample_rate and its alias colsample_bylevel are both set");
```

Comment:
I've got a recommendation from @michalkurka to check this error using this way in a similar test in other PR, so I use it also now.  The reason is, we should check it fails with the correct message because, for example, someone can set some wrong parameter and it fails on something different earlier.

Reply:
@maurever yes, correct - we should always check the type of exception and expected error message

this can be done in 2 ways 
- by catching the expected exception



- by using ExpectedException and expectMessage

it is up to the developer which one to chose (both have advantages)

this particular test needs improvement:
- use assertTrue instead of assert
- make sure the test fails if no exception is thrown, eg. add fail() to line 2735

Reply:
@michalkurka, fixed here: https://github.com/h2oai/h2o-3/pull/5641/commits/fe2b5287bbbb141a5d88be32798b0a6484463649